journal.py

```diff
@@ -106,13 +100,15 @@ def edit_entry(request):
         try:
             entry = Entry.search(post_id)
         except NoResultFound:
-            return HTTPNotFound('No post found.')
+            return HTTPNotFound('There is no post with this id.')
         if request.method == 'POST':
-            id = request.matchdict['id']
-            title = request.params.get('title')
-            text = request.params.get('text')
-            Entry.update(id=id, title=title, text=text)
-            return HTTPFound(request.route_url('home'))
+            entry.title = request.params.get('title')
+            entry.text = request.params.get('text')
+            if 'HTTP_X_REQUESTED_WITH' not in request.environ:
+                return HTTPFound(request.route_url('detail', id=post_id))
+            else:
+                entry_dict = {'title': entry.title}
+                return entry_dict
```

Comment:
I'm curious about what happens here.  Looking forward to reviewing the js end of it.